core/struct.rbs

```diff
@@ -1,299 +1,51 @@
-  #
-  # By default, the arguments for initializing an instance of the new subclass can
-  # be both positional and keyword arguments.
-  #
-  # Optional keyword argument `keyword_init:` allows to force only one type of
-  # arguments to be accepted:
-  #
-  #     KeywordsOnly = Struct.new(:foo, :bar, keyword_init: true)
-  #     KeywordsOnly.new(bar: 1, foo: 0)
-  #     # => #<struct KeywordsOnly foo=0, bar=1>
-  #     KeywordsOnly.new(0, 1)
-  #     # Raises ArgumentError: wrong number of arguments
-  #
-  #     PositionalOnly = Struct.new(:foo, :bar, keyword_init: false)
-  #     PositionalOnly.new(0, 1)
-  #     # => #<struct PositionalOnly foo=0, bar=1>
-  #     PositionalOnly.new(bar: 1, foo: 0)
-  #     # => #<struct PositionalOnly foo={:foo=>1, :bar=>2}, bar=nil>
-  #     # Note that no error is raised, but arguments treated as one hash value
-  #
-  #     # Same as not providing keyword_init:
-  #     Any = Struct.new(:foo, :bar, keyword_init: nil)
-  #     Any.new(foo: 1, bar: 2)
-  #     # => #<struct Any foo=1, bar=2>
-  #     Any.new(1, 2)
-  #     # => #<struct Any foo=1, bar=2>
-  #
-  def initialize: (attribute_name, *attribute_name, ?keyword_init: boolish) ?{ () -> void } -> void
+  def self.members: () -> Array[Symbol]
+  def self.keyword_init?: () -> bool?
 
-  # <!--
-  #   rdoc-file=struct.c
-  #   - each {|value| ... } -> self
-  #   - each -> enumerator
-  # -->
-  # Calls the given block with the value of each member; returns `self`:
-  #
-  #     Customer = Struct.new(:name, :address, :zip)
-  #     joe = Customer.new("Joe Smith", "123 Maple, Anytown NC", 12345)
-  #     joe.each {|value| p value }
-  #
-  # Output:
-  #
-  #     "Joe Smith"
-  #     "123 Maple, Anytown NC"
-  #     12345
-  #
-  # Returns an Enumerator if no block is given.
-  #
-  # Related: #each_pair.
-  #
-  def each: () -> ::Enumerator[Elem?, self]
-          | () { (Elem? item) -> void } -> self
+  def ==: (untyped other) -> bool
+  def eql?: (untyped other) -> bool
+  def hash: () -> Integer
 
-  # <!--
-  #   rdoc-file=struct.c
-  #   - StructClass::members -> array_of_symbols
-  # -->
-  # Returns the member names of the Struct descendant as an array:
-  #
-  #     Customer = Struct.new(:name, :address, :zip)
-  #     Customer.members # => [:name, :address, :zip]
-  #
-  def self.members: () -> ::Array[Symbol]
+  def inspect: () -> String
+  alias to_s inspect
 
-  # <!--
-  #   rdoc-file=struct.c
-  #   - StructClass::keyword_init? -> true or falsy value
-  # -->
-  # Returns `true` if the class was initialized with `keyword_init: true`.
-  # Otherwise returns `nil` or `false`.
-  #
-  # Examples:
-  #     Foo = Struct.new(:a)
-  #     Foo.keyword_init? # => nil
-  #     Bar = Struct.new(:a, keyword_init: true)
-  #     Bar.keyword_init? # => true
-  #     Baz = Struct.new(:a, keyword_init: false)
-  #     Baz.keyword_init? # => false
-  #
-  def self.keyword_init?: () -> (true | false | nil)
+  def to_a: () -> Array[Elem?]
```

Comment:
does this have the same signature as the `#to_a` from `Enumerable[Elem?]`?

Reply:
No. Since  you can do `My3dPoint.new(1)`, the extra elements are assigned as `nil`. Thus, anything that interacts with them are `Elem?`.

Reply:
:arrow_right: `#select` comment https://github.com/ruby/rbs/pull/1456#discussion_r1306588308